e2e/testcases/csr_auth_test.go

```diff
@@ -162,64 +162,62 @@ func TestFleetWIDifferentProject(t *testing.T) {
 	})
 }
 
-// getMembershipIdentityProvider fetches the workload_identity_pool if the membership exists.
-func getMembershipIdentityProvider(nt *nomostest.NT) (bool, string, error) {
-	bytes, err := nt.Kubectl("get", "membership", "membership")
+// getMembership gets the membership resource in the GCP project.
+// It returns a boolean that indicates whether the membership exists, and an error
+func getMembership(fleetMembership, gcpProject string) (bool, error) {
+	bytes, err := exec.Command("gcloud", "container", "fleet", "memberships", "describe", fleetMembership, "--project", gcpProject).CombinedOutput()
 	out := string(bytes)
 	if err != nil {
-		if strings.Contains(out, `the server doesn't have a resource type "membership"`) || strings.Contains(out, "NotFound") {
-			return false, "", nil
+		// ERROR: (gcloud.container.fleet.memberships.describe) Membership xxx not found in the fleet.
+		if strings.Contains(out, "not found") {
```

Comment:
Seems like this could catch other similar errors. Maybe check for soemthing more exact, like `Membership %s not found in the fleet.` with Sprintf?

Reply:
Sure. My concern is that the message is from Hub controller and might change.

I guess it is okay to update the string if it changes in the future.

DONE